### DISCUSSION TITLE
URL Structure to drop Hash Fragments

### DISCUSSION THREAD

**Discussion by giliomeejg** [gitcoinco/grants-stack#969] [Ideas]
The current URL structure uses hash fragments, e.g.
[https://grant-explorer.gitcoin.co/#/round/1/0xd..](https://grant-explorer.gitcoin.co/#/round/1/0xdf75054cd67217aee44b4f9e4ebc651c00330938)

The rationale behind considering an alternative structure:

**1). Google Analytics 4 doesn't support Hash Fragments.**  
MMM is currently looking into a customized solution to solve for this and in the process of testing, however there's concern around creating unnecessary complexity.

**2). The impact on SEO**
[https://developers.google.com/..](https://developers.google.com/search/blog/2020/05/frequently-asked-questions-about)

> Does Googlebot understand fragment URLs? 
> Fragment URLs, also known as "hash URLs", are technically fine, but might not work the way you expect with Googlebot.
> 
> Fragments are supposed to be used to address a piece of content within the page and when used for this purpose, fragments are absolutely fine.
> 
> Sometimes developers decide to use fragments with JavaScript to load different content than what is on the page without the fragment. That is not what fragments are meant for and won't work with Googlebot. See the JavaScript SEO guide on how the History API can be used instead.
> 

As we're hosting via Fleek, routing correctly is somewhat problematic, however I think I've found a solution to this, and propose that we consider changing the URL structure to:
[https://grant-explorer.gitcoin.co/round/1/0xd..](https://grant-explorer.gitcoin.co/round/1/0xdf75054cd67217aee44b4f9e4ebc651c00330938)

**A potential solution**

In order to implement this via Fleek, we could use [IPFS _redirects](https://github.com/ipfs/specs/blob/1ac9290c52cb351462ddb6fe5c1e5fb809613636/http-gateways/REDIRECTS_FILE.md) to effectively setup rewriting rules for the routes we're creating and load the index.html file.

I've done a test, hosted on Fleek for [https://nudge.tech/about](https://nudge.tech/about) which seems to work fine.

Related: #2189

**Comment by vacekj** [comment]
@nfrgosselin is this something we still need to address?

**Reply by giliomeejg** [reply]
Yes.  Would love to jump on a call if it needs to be explored further.
